src/basic/user-util.c

```diff
@@ -350,6 +350,9 @@ char* uid_to_name(uid_t uid) {
                         if (r != ERANGE)
                                 break;
 
+                        if (bufsize*2 < bufsize) /* overflow check */
+                                return NULL;
```

Comment:
signed overflow is undefined... So this check is not really valid. If we go full-paranoia mode, then I think we need `bufsize > LONG_MAX / 2`.

Reply:
Fix this part though please.